addons/controllers/secretgen-controller/suite_test.go

```diff
@@ -0,0 +1,162 @@
+		ErrorIfCRDPathMissing: true,
+	}
+
+	externalCRDPaths, err := getExternalCRDPaths()
+	Expect(err).NotTo(HaveOccurred())
+	Expect(externalCRDPaths).ToNot(BeEmpty())
+	testEnv.CRDDirectoryPaths = externalCRDPaths
+	testEnv.CRDDirectoryPaths = append(testEnv.CRDDirectoryPaths, filepath.Join("..", "..", "..", "config", "crd", "bases"))
+	testEnv.ErrorIfCRDPathMissing = true
+
+	cfg, err = testEnv.Start()
+	Expect(err).ToNot(HaveOccurred())
+	Expect(cfg).ToNot(BeNil())
+
+	err = clientgoscheme.AddToScheme(scheme)
+	Expect(err).NotTo(HaveOccurred())
+
+	err = clusterapiv1beta1.AddToScheme(scheme)
+	Expect(err).NotTo(HaveOccurred())
+
+	err = controlplanev1beta1.AddToScheme(scheme)
+	Expect(err).NotTo(HaveOccurred())
+
+	// Include config API
+	err = addonsv1alpha1.AddToScheme(scheme)
+	Expect(err).NotTo(HaveOccurred())
+
+	k8sClient, err = client.New(cfg, client.Options{Scheme: scheme})
+	Expect(err).ToNot(HaveOccurred())
+	Expect(k8sClient).ToNot(BeNil())
+
+	dynamicClient, err = dynamic.NewForConfig(cfg)
+	Expect(err).ToNot(HaveOccurred())
+	Expect(dynamicClient).ToNot(BeNil())
+
+	options := manager.Options{
+		Scheme:             scheme,
+		MetricsBindAddress: "0",
+		Port:               9443,
+	}
+	mgr, err := ctrl.NewManager(testEnv.Config, options)
+	Expect(err).ToNot(HaveOccurred())
+
+	setupLog := ctrl.Log.WithName("controllers").WithName("SecretGenControllerConfig")
+
+	ctx, cancel = context.WithCancel(ctx)
+
+	Expect((&SecretGenControllerConfigReconciler{
+		Client: mgr.GetClient(),
+		Log:    setupLog,
+		Scheme: mgr.GetScheme(),
+	}).SetupWithManager(ctx, mgr, controller.Options{MaxConcurrentReconciles: 1})).To(Succeed())
+
+	// pre-create namespace
+	ns := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "tkr-system"}}
+	Expect(k8sClient.Create(context.TODO(), ns)).To(Succeed())
+
+	go func() {
+		defer GinkgoRecover()
+		Expect(mgr.Start(ctx)).To(Succeed())
+	}()
+
+	close(done)
+}, 60)
+
+var _ = AfterSuite(func() {
+	By("tearing down the test environment")
+	cancel()
+	err := testEnv.Stop()
+	Expect(err).ToNot(HaveOccurred())
+})
+
+// get paths for external CRDs by introspecting versions of the go dependencies
+func getExternalCRDPaths() ([]string, error) {
+	externalDeps := map[string][]string{
+		"sigs.k8s.io/cluster-api": {"config/crd/bases",
+			"controlplane/kubeadm/config/crd/bases"},
+	}
+
+	var crdPaths []string
+	gopath, err := exec.Command("go", "env", "GOPATH").Output()
+	if err != nil {
+		return crdPaths, err
+	}
+	for dep, crdDirs := range externalDeps {
+		depPath, err := exec.Command("go", "list", "-m", "-f", "{{ .Path }}@{{ .Version }}", dep).Output()
+		if err != nil {
+			return crdPaths, err
+		}
+		for _, crdDir := range crdDirs {
+			crdPaths = append(crdPaths, filepath.Join(strings.TrimSuffix(string(gopath), "\n"),
+				"pkg", "mod", strings.TrimSuffix(string(depPath), "\n"), crdDir))
+		}
+	}
+
+	logf.Log.Info("external CRD paths", "crdPaths", crdPaths)
+	return crdPaths, nil
+}
```

Comment:
If we move up the controllers for Config API objects to the same level as the addons controller, we don't need to rewrite the same suite_test.go over all subdirectories. Maybe something to consider